packages/edit-site/src/components/editor/index.js

```diff
@@ -133,6 +132,7 @@ export default function Editor( { listViewToggleElement, isLoading } ) {
 				: undefined,
 			editorMode: getEditorMode(),
 			canvasMode: getCanvasMode(),
+			renderingMode: getRenderingMode(),
```

Comment:
We have so many modes: canvasMode, editorMode, renderingMode. To be honest, I don't like that and I'm not sure how to better name this. I'm opening to changing the name. Could be `contentMode` or `focusMode` or `contentFocusMode` or something. I'm not sure, but I know that focusMode already exists for something else.

Reply:
There are two modes that use "focus" as far as I know:

1. "Spotlight mode" in preferences (`focusMode` variable)
2. Focus mode is also used to describe how the editor behaves for navigation, patterns etc with resize handles. It's not a great name I think, well, it left me scratching my head at least. Reference: https://github.com/WordPress/gutenberg/issues/55122#issuecomment-1752218273

Yeah, the challenge for managing this state is that most, if not all, of these modes are not mutually exclusive - there are various combinations.

An editor could be in "zoom-out" mode (`__unstableGetEditorMode`) and "template-only" (`renderingMode`), `is-outline-mode` (postType === 'wp_template').

Then canvas mode, navigation/edit mode and whatever else.